Examples/OISTVisualizationTool/main.swift

```diff
@@ -265,6 +265,278 @@ struct PpcaTrack: ParsableCommand {
   }
 }
 
+/// Returns a tracking configuration for a tracker using an RAE.
+///
+/// Parameter model: The RAE model to use.
+/// Parameter statistics: Normalization statistics for the frames.
+/// Parameter frames: The frames of the video where we want to run tracking.
+/// Parameter targetSize: The size of the target in the frames.
+public func makeNaiveBayesRAETracker(
+  model: DenseRAE,
+  statistics: FrameStatistics,
+  frames: [Tensor<Double>],
+  targetSize: (Int, Int),
+  foregroundModel: GaussianNB,
+  backgroundModel: GaussianNB
+) -> TrackingConfiguration<Tuple1<Pose2>> {
+  var variableTemplate = VariableAssignments()
+  var frameVariableIDs = [Tuple1<TypedID<Pose2>>]()
+  for _ in 0..<frames.count {
+    frameVariableIDs.append(
+      Tuple1(
+        variableTemplate.store(Pose2())
+        ))
+  }
+  return TrackingConfiguration(
+    frames: frames,
+    variableTemplate: variableTemplate,
+    frameVariableIDs: frameVariableIDs,
+    addPriorFactor: { (variables, values, graph) -> () in
+      let (poseID) = unpack(variables)
+      let (pose) = unpack(values)
+      graph.store(WeightedPriorFactor(poseID, pose, weight: 1e-2))
+    },
+    addTrackingFactor: { (variables, frame, graph) -> () in
+      let (poseID) = unpack(variables)
+      graph.store(
+        ProbablisticTrackingFactor(poseID,
+          measurement: statistics.normalized(frame),
+          encoder: model,
+          patchSize: targetSize,
+          appearanceModelSize: targetSize,
+          foregroundModel: foregroundModel,
+          backgroundModel: backgroundModel
+        )
+      )
+    },
+    addBetweenFactor: { (variables1, variables2, graph) -> () in
+      let (poseID1) = unpack(variables1)
+      let (poseID2) = unpack(variables2)
+      graph.store(WeightedBetweenFactor(poseID1, poseID2, Pose2(), weight: 1e-2))
+    })
+}
+
+/// The dimension of the hidden layer in the RAE appearance model.
+let kHiddenDimension = 100
+
+/// The dimension of the latent code in the RAE appearance model.
+let kLatentDimension = 10
+
+/// Tracking with a Naive Bayes with RAE
+struct NaiveRae: ParsableCommand {
+  @Option(help: "Location of dataset folder which should contain `frames` and `frames_txt`")
+  var datasetLocation: String = "./OIST_Data"
+
+  @Option(help: "Where to load the RAE weights")
+  var loadWeights: String = "./oist_rae_weight.npy"
+
+  @Option(help: "Which bounding box to track")
+  var boxId: Int = 0
+
+  @Option(help: "Track for how many frames")
+  var trackFrames: Int = 10
+
+  @Flag(help: "Print progress information")
+  var verbose: Bool = false
+
+  /// Returns predictions for `videoName` using the raw pixel tracker.
+  func naiveRaeTrack(dataset dataset_: OISTBeeVideo, length: Int, ppcaSize: Int = 10, ppcaSamples: Int = 100) -> [OrientedBoundingBox] {
```

Comment:
this function is way too long, and poorly documented at that. Kill two birds w one stone by splitting into some documented smaller functions?